### ISSUE TITLE
Additional USMT XML files, save more settings

### ISSUE THREAD

**Issue by Myria-de** [belowaverage-org/SuperGrate#107]
**Is your feature request related to a problem? Please describe.**
Super Grate saves too few settings.

**Describe the solution you'd like**
Selection of additional USMT XML files, take more settings into account.

**Describe alternatives you've considered**
A menu that can be used to select additional XML files or Apps.

**Additional context**
I also prefer to keep a program as simple as possible. With USMT, however, you should perhaps be more specific about what is saved and restored from the profile. Perhaps an optional selection of XML files should be offered so that more settings are saved. Current files can be downloaded from https://ehlertech.com/customxmls/, for example. The MigApp.xml file from them also takes the Firefox profile of current versions into account. This would certainly make Super Grate useful for more users.

Super Grate is probably aimed more at experienced users. Of course, you can also copy additional user defined XML files to the "USMT\X64" folder and specify them after an additional "/i" in the ScanStateParameters and LoadStateParameters settings.

**Comment by Vorax1213** [comment]
I agree with this request, especially the additional part where we should be able to see what is actually saved and restored from the profile (size + content). Thank you for your great work already !

**Comment by krisdb2009** [comment]
Looks like someone added firefox support in another fork, I pulled it in:

https://github.com/belowaverage-org/SuperGrate/pull/129

**Comment by ISU-Ncwater** [comment]
> Looks like someone added firefox support in another fork, I pulled it in:
> 
> [#129](https://github.com/belowaverage-org/SuperGrate/pull/129)

Hey, its been a while and actually forgot I was working on this. My previous version of Firefox support doesn't work as intended, as it is different than Chrome and Edge. I've just tested this new version and it worked for me on a local copy. The scan wasn't detecting and Firefox folders/files before. Additionally, there are errors in the Chrome and Edge sections that have "UserData" instead of "User Data" causing the file to do nothing for those browsers. Here is the updated section of Firefox within MigSgAdditional.xml

```
`<component context="User" type="Application">
    <displayName>Mozilla Firefox</displayName>
    <role role="Settings">
        <rules context="User">
            <include>
                <objectSet>
                    <!-- Capture entire Firefox profile structure -->
                    <pattern type="File">%CSIDL_APPDATA%\Mozilla\Firefox\* [*]</pattern>
                </objectSet>
            </include>
        </rules>
    </role>
</component>`
```
